### PR TITLE
[$100 bounty] mapsd: switch to static render mode

### PR DESCRIPTION
**Pull request by ksfi**
Concerning bounty #31333 

Changes continuous rendering to static rendering.

The style.json file was returning 404 errors from the MapBox API so I used the url style back.

It also needs more frames to pass from valid to invalid, maybe due to using the full style map.

### CODE REVIEW

File: selfdrive/navd/map_renderer.cc [hunk 1]
```diff
@@ -62,7 +64,7 @@ MapRenderer::MapRenderer(const QMapLibre::Settings &settings, bool online) : m_s
   std::string style = util::read_file(STYLE_PATH);
   m_map.reset(new QMapLibre::Map(nullptr, m_settings, fbo->size(), 1));
   m_map->setCoordinateZoom(QMapLibre::Coordinate(0, 0), DEFAULT_ZOOM);
-  m_map->setStyleJson(style.c_str());
```
Comment (adeebshihadeh) [inline selfdrive/navd/map_renderer.cc:65]: why doesn't this work?

File: selfdrive/navd/map_renderer.h [hunk 2]
```diff
@@ -38,13 +38,16 @@ class MapRenderer : public QObject {
   void publish(const double render_time, const bool loaded);
   void sendThumbnail(const uint64_t ts, const kj::Array<capnp::byte> &buf);
 
+  //QMapLibre::Style style;
   QMapLibre::Settings m_settings;
   QScopedPointer<QMapLibre::Map> m_map;
 
   void initLayers();
 
+  double start_render_t = 0, end_render_t = 0;
```
Comment (adeebshihadeh) [inline selfdrive/navd/map_renderer.h:47]: `end_render_t` doesn't need to be kept around with the class

File: selfdrive/navd/map_renderer.h [hunk 3]
```diff
@@ -38,13 +38,16 @@ class MapRenderer : public QObject {
...
   uint32_t frame_id = 0;
   uint64_t last_llk_rendered = 0;
+  bool ok_render = true;
```
Comment (adeebshihadeh) [inline selfdrive/navd/map_renderer.h:50]: can you give this a more descriptive name? had to read around to figure out what this was for

File: selfdrive/navd/tests/test_map_renderer.py [hunk 4]
```diff
@@ -135,7 +135,7 @@ def _run_test(self, expect_valid, locations=LOCATION1_REPEATED):
       invalid_and_not_previously_valid = (expect_valid and not self.sm.valid['mapRenderState'] and not prev_valid)
       valid_and_not_previously_invalid = (not expect_valid and self.sm.valid['mapRenderState'] and prev_valid)
 
-      if (invalid_and_not_previously_valid or valid_and_not_previously_invalid) and frames_since_test_start < 5:
+      if (invalid_and_not_previously_valid or valid_and_not_previously_invalid) and frames_since_test_start < 20:
```
Comment (adeebshihadeh) [inline selfdrive/navd/tests/test_map_renderer.py:138]: why did this change?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: selfdrive/navd/map_renderer.cc (hunk 1 above)
Comment (ksfi) [reply selfdrive/navd/map_renderer.cc]: actually it's working fine, the 404 errors were coming from the number of frames needed to pass from valid to invalid and vice-versa (below)

File: selfdrive/navd/tests/test_map_renderer.py (hunk 4 above)
Comment (ksfi) [reply selfdrive/navd/tests/test_map_renderer.py:138]: might come from static rendering asynchrony
from tests and trials it needs between 15 and 20 frames to pass
it not being an assertion i assumed i could modify it

**Pull request merged**